core/nakamoto/mempool.go

```diff
@@ -9,6 +17,9 @@ package nakamoto
 //
 // Note that due to how Nakamoto consensus works, there is the possibility of reorgs, which means that a block that was previously mined may be replaced by a longer chain. In this case, transactions which have been taken from the mempool and included in a block that is later reorged out should be "returned" to the mempool. This is the intuition for the mempool's behaviour, however it is designed as a one-way flow.
 type Mempool struct {
```

Comment:
Posting link here to track our designs

https://www.tldraw.com/r/hFaij_Nw3R9jdJdAO038p?v=-394,-21,1865,970&p=page